scylla-persistence/src/main/java/com/netflix/conductor/scylla/dao/ScyllaExecutionDAO.java

```diff
@@ -331,38 +342,67 @@ public List<TaskModel> createTasks(List<TaskModel> tasks) {
      * @method to add the task_in_progress table with the status of the task if task is not already present
      */
     public void addTaskInProgress(TaskModel task) {
-        ResultSet resultSet =
-                session.execute(
-                        selectTaskInProgressStatement.bind(task.getTaskDefName(),
-                                UUID.fromString(task.getTaskId())));
-        if (resultSet.all().isEmpty() || resultSet.all().size()<1) {
-            session.execute(
-                    insertTaskInProgressStatement.bind(task.getTaskDefName(),
-                            UUID.fromString(task.getTaskId()),
-                            UUID.fromString(task.getWorkflowInstanceId()),
-                            true));
+        ResultSet resultSet = getTaskInProgressTaskFromTaskModel(task);
+        if (CollectionUtils.isEmpty(resultSet.all())) {
+            session.execute(insertTaskInProgressV2Statement.bind(task.getTaskDefName(), UUID.fromString(task.getTaskId()),
+                    UUID.fromString(task.getWorkflowInstanceId()), true));
+        } else {
+            LOGGER.info("Task with defName {} and Id {} and status {} in addTaskInProgress NOT inserted as already exists  ", task.getTaskDefName(),
+                    task.getTaskId(), task.getStatus());
         }
-        else {
-            LOGGER.info("Task with defName {} and Id {} and status {} in addTaskInProgress NOT inserted as already exists  "
-                    ,task.getTaskDefName(), task.getTaskId(),task.getStatus());
+    }
+
+    private ResultSet getTaskInProgressTaskFromTaskModel(TaskModel task) {
+        try {
+            ResultSet resultSet = fetchFromTaskInProgressV2Query(task);
+            if (isResultSetNotValid(resultSet)) {
+                return fetchFromTaskInProgressQuery(task);
```

Comment:
Do we need to add any metric, how many times we are fetching from V2 vs V1

Reply:
we will read and write in both till we switch off the flag, after that all read and write directs to new table